docs/dev/guidelines/server-tests.rst

```diff
@@ -178,5 +178,44 @@ We no longer mock the uppermost method but only throw the exception at the place
 For a real example where we replaced a SpyBean with a static mock, look at the ``SubmissionExportIntegrationTest.java`` `here <https://github.com/ls1intum/Artemis/commit/4843137aa01cfdf27ea019400c48df00df36ed45>`__.
 
 
+7. Parallel test execution
+===========================
+We use the `JUnit 5 <https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution>`__ feature to execute tests in parallel.
 
+The following line in the ``junit-platform.properties`` file enables parallel test execution with JUnit 5. Setting the property to ``false`` disables parallel test execution.
 
+.. code-block:: properties
+
+    junit.jupiter.execution.parallel.enabled = true
+
+To execute a test class and its inheriting classes in parallel, annotate it with ``@Execution(ExecutionMode.CONCURRENT)``. Since we need to isolate resources such as
```

Comment:
The current reading flow is alternating

- ResourceLock
- Isolated
- ResourceLock
- Isolated.

This makes it somewhat hard to follow.

Could you maybe start a small section only for resource locks, and then another one for isolated? That would make the structure and intention more understandable:
- by default please use `@Execution(ExecutionMode.CONCURRENT)`
- if not possible, a resource lock still allows limited concurrency, and
- if that is not possible either, use `@Isolated`.

```suggestion
To execute a test class and its inheriting classes in parallel, annotate it with ``@Execution(ExecutionMode.CONCURRENT)``.

ResourceLock
''''''''''''
Since we need to isolate resources such as […]


Isolated
''''''''
[…]
```

Reply:
I agree, maybe one addition would be to list example tests directly from Artemis so that people can look directly in the codebase to see the differences.